src/App.tsx

```diff
@@ -1,26 +1,152 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Errors } from './components/Errors';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorMessage } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredStatus, setFilteredStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [tempoTodo, setTempoTodo] = useState<Todo | null>(null);
+  const [title, setTitle] = useState('');
+  const [deletedTodo, setDeletedTodo] = useState(NaN);
+  const [isDeleteCompleted, setIsDeleteCompleted] = useState(false);
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => setErrorMessage(''), 3000);
+
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToLoad);
+        clearTimeout(timeoutId);
+      });
+
+    return () => clearTimeout(timeoutId);
+  }, []);
+
+  const handleAdd = useCallback((newTodo: Todo) => {
+    setTempoTodo(newTodo);
+    let todosLength = 0;
+
+    addTodo(newTodo)
+      .then(todo => {
+        setTodos(currentTodos => [...currentTodos, todo]);
+        todosLength = 1;
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToAdd);
+      })
+      .finally(() => {
+        if (todosLength === 1) {
+          setTitle('');
+        }
+
+        setTempoTodo(null);
+      });
+  }, []);
+
+  const handleDeleteCompleted = () => {
+    setIsDeleteCompleted(true);
+    const completedTodos = todos.filter(todo => todo.completed);
+
+    Promise.allSettled(
+      completedTodos.map(todo => deleteTodo(todo.id).then(() => todo)),
+    )
+      .then(values => {
+        values.forEach(val => {
+          if (val.status === 'rejected') {
+            setErrorMessage(ErrorMessage.UnableToDelete);
+          } else {
+            setTodos(currentTodos => {
+              const todoId = val.value as Todo;
+
+              return currentTodos.filter(todo => todo.id !== todoId.id);
+            });
+          }
+        });
+      })
+      .finally(() => setIsDeleteCompleted(false));
+  };
+
+  const handleDelete = (todoId: number) => {
+    setDeletedTodo(todoId);
+
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessage.UnableToDelete);
+      })
+      .finally(() => setDeletedTodo(NaN));
```

Comment:
i think will be better set null
```suggestion
      .finally(() => setDeletedTodo(null));
```